flycheck.el

```diff
@@ -9353,8 +9353,9 @@ See URL `http://pear.php.net/package/PHP_CodeSniffer/'."
             ;; Some files are not detected correctly
             ;; so it is necessary to pass the extension.
             (eval
-             (when (buffer-file-name)
-               (concat "--extensions=" (file-name-extension (buffer-file-name)))))
+             (-when-let (ext (file-name-extension buffer-file-name))
+               (concat "--extensions=" (file-name-extension buffer-file-name))))
```

Comment:
Sorry, this is still wrong.  You reintroduced the same problem: if buffer-file-name is nil, this will throw an exception.

Reply:
Did you test this code? It sets the variable `ext` to the value of `buffer-file-name`